### PR TITLE
clientv3/concurrency: preallocate slice in stm

### PR DESCRIPTION
**Pull request by gyuho**
/cc @heyitsanthony @xiang90 

Thanks!

### CODE REVIEW

File: clientv3/concurrency/stm.go [hunk 1]
```diff
@@ -201,12 +203,16 @@ func (s *stmSerializable) Rev(key string) int64 {
 	return s.stm.Rev(key)
 }
 
-func (s *stmSerializable) gets() (keys []string, ops []v3.Op) {
+func (s *stmSerializable) gets() ([]string, []v3.Op) {
+	var (
```
Comment (heyitsanthony) [inline clientv3/concurrency/stm.go]: I don't understand the advantage of doing this over :=

File: clientv3/client.go [hunk 2]
```diff
@@ -153,10 +153,11 @@ func (cred authTokenCredential) GetRequestMetadata(ctx context.Context, s ...str
 
 // Dial establishes a connection for a given endpoint using the client's config
 func (c *Client) Dial(endpoint string) (*grpc.ClientConn, error) {
-	opts := []grpc.DialOption{
+	opts := make([]grpc.DialOption, 0, 5)
```
Comment (heyitsanthony) [inline clientv3/client.go]: probably not worth doing this one since it's rarely called and would have to be updated if new dial options are added

**Comment by gyuho** [comment]
@heyitsanthony All addressed. PTAL.

Thanks!

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**